### ISSUE TITLE
Got error[E0227] when compile amq-proto

### ISSUE THREAD

**Issue by Qinka** [Antti/rust-amq-proto#2]
I got an error(E0227).

```
error[E0277]: the trait bound `error::Error: std::convert::From<std::io::Error>` is not satisfied
   --> C:\Users\qinka\.cargo\registry\src\mirrors.ustc.edu.cn-61ef6e0cd06fb9b8\amq-proto-0.1.0\src\codegen_macros.rs:148:23
    |
148 |             let res = self.data.write_all(&self.bits.to_bytes()).map_err(From::from);
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::convert::From<std::io::Error>` is not implemented for `error::Error`
    |
    = help: the following implementations were found:
              <error::Error as std::convert::From<&'a str>>
              <error::Error as std::convert::From<error::ErrorKind>>
              <error::Error as std::convert::From<std::string::String>>
    = note: required by `std::convert::From::from`
```

The version of Rust compiler is
```
rustc 1.34.0-nightly (3315728c0 2019-02-09)
```

**Comment by atoav** [comment]
I get the same error with `1.33`, but for me it looks more like this (scroll to line 359 to see it starting): https://ci.appveyor.com/project/atoav/bender-worker/build/job/de75trbhis6tjpvc

I want to add, that this seems to only happen on windows targets, not on unix targets. It would be quite cool if this could be somehow resolved..

**Comment by atoav** [comment]
I could reproduce this error locally on a Windows system by cloning this reprository and running `cargo build`. Weirdly enough it builds perfectly fine on unix based systems. On Windows it fails with:  
```
error: Could not compile `amq-proto`.
warning: build failed, waiting for other jobs to finish...
error: aborting due to 108 previous errors 
```
I checked and it seems that all these 108 Errors seem to be related to Error handling, many of them obviously caused by another. Except for two errors in `src/framing.rs` all other errors happen in `src/table.rs` and `src/codegen_macros.rs`.  I am not at all familiar with macros in Rust, but I suspect the Macros are the root cause?

I am working on a cross plattform project that uses `rust-amqp`. This seems to be the onlyt remaining issue that stops a Windows build for me, it would be extremely cool if this would be resolved.

**Comment by Qinka** [comment]
@atoav ,  I used [lapin](https://github.com/sozu-proxy/lapin) instead.

**Comment by atoav** [comment]
> @atoav , I used [lapin](https://github.com/sozu-proxy/lapin) instead.  

Thanks for the suggestion, but I'd like to avoid it, because my application relies in many places on the rust-amqp Channel, and it would be a kinda length operation, that I am not really prepared to go through right now..

**Comment by tyrylu** [comment]
Same there, i wanted to use amqp, because lapin is behaving somewhat weirdly with async/await, mainly the consumption, so i wanted to try something without futures...

**Comment by Qinka** [comment]
@tyrylu  Believe me. You can find a way out with lapin, even it's weird. However, the only way you can find in this repo is named "no way".

**Comment by tyrylu** [comment]
Yeah, it seems so. But i have a weird lapin_futures issue as well, see https://github.com/sozu-proxy/lapin/issues/165

**Comment by zhendexuebuhui** [comment]
I also encountered this problem

**Comment by StyMaar** [comment]
This error happens because error chain (or at least version 0.10) only implement the conversion needed conversion for Unix platforms:

```
    #[cfg(unix)]
    impl From<::std::io::Error> for Error {
        fn from(e: ::std::io::Error) -> Self {
            Error::from_kind(ErrorKind::Io(e))
        }
    }
```

(This is the output of [cargo-expand](https://github.com/dtolnay/cargo-expand) for the error module of this crate). That's pretty unfortunate, but I guess everybody already moved away from this library so I must be the last person affected…

**Comment by keyurnpatel** [comment]
encountered same issue.
